app/uk/gov/hmrc/gform/gform/EnrolmentController.scala

```diff
@@ -270,6 +289,39 @@ class EnrolmentController(
               }
     )
 
+  def inSufficientCredentialsPageSubmit(formTemplateId: FormTemplateId) =
+    withEnrolmentSection(formTemplateId)(formTemplate =>
+      enrolmentSection =>
+        enrolmentOutcomes =>
+          implicit sse =>
+            implicit request =>
+              implicit l =>
+                changeOrSignoutChoice
+                  .bindFromRequest()
+                  .fold(
+                    errorForm => {
+                      val insufficientCredentialsPage = new InsufficientCredentialsPage(formTemplate, errorForm)
+                      BadRequest(
+                        insufficient_credentials_page(
+                          insufficientCredentialsPage,
+                          enrolmentOutcomes.insufficientCredentialsPage,
+                          frontendAppConfig
+                        )
+                      )
+                    },
+                    {
+                      case "change-gg-account" =>
+                        Redirect(
+                          uk.gov.hmrc.gform.gform.routes.NewFormController.dashboardWithNewSession(formTemplate._id)
+                        )
+                      case "sign-out" =>
+                        Redirect(uk.gov.hmrc.gform.gform.routes.SignOutController.signOut(formTemplate._id))
+                      case unknown =>
+                        throw new Exception(s"Unexpected value of linked account parameter: '$unknown'")
+                    }
```

Comment:
This is now duplicated twice, can we extract it into a common method.

Reply:
Done